src/libsync/propagateupload.cpp

```diff
@@ -190,22 +194,53 @@ bool PollJob::finished()
     return true;
 }
 
-
 void PropagateUploadFileQNAM::start()
 {
     if (_propagator->_abortRequested.fetchAndAddRelaxed(0)) {
         return;
     }
 
+    const QString filePath = _propagator->getFilePath(_item._file);
+
+    // remember the modtime before checksumming to be able to detect a file
+    // change during the checksum calculation
+    _item._modtime = FileSystem::getModTime(filePath);
+
+    _stopWatch.start();
+
+    // do whatever is needed to add a checksum to the http upload request.
+    // in any case, the validator will emit signal startUpload to let the flow
+    // continue in slotStartUpload here.
+    _validator = new TransmissionChecksumValidator(filePath);
+    connect(_validator, SIGNAL(validated()), this, SLOT(slotStartUpload()));
```

Comment:
_validator is neither initialized to null, nor set to null after being deleted. This could end up causing a crash as the code evolve.

You also don't seem to need to keep the pointer, you could set the PropagateUploadFileQNAM as it's QObject parent (just to for avoid leaking in edge cases) and connect its deleteLater() slot to validated() as well.